### ISSUE TITLE
Lint request: warn against drop(manually_drop)

### ISSUE THREAD

**Issue by nico-abram** [rust-lang/rust-clippy#5581]
I just had the thought that maybe clippy should lint against this kind of thing: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1e59503ba3998375e9f33689bcee139b

I think someone might find it confusing that calling drop on it doesn't actually drop the inner value (Which should be dropped using the unsafe drop method on it)

**Comment by flip1995** [comment]
```rust
struct S;

impl Drop for S {
    fn drop(&mut self) {
        println!("drip drop");
    }
}

fn main() {
    drop(std::mem::ManuallyDrop::new(S));
    unsafe {
        std::mem::ManuallyDrop::drop(&mut std::mem::ManuallyDrop::new(S));
    }
}
```

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7cd9652d0fa6328c574e6ee5f20d5e74)

**Comment by cgm616** [comment]
I'd like to take this one!
